File: services/api/app/lib/client_facing_safety.py
from __future__ import annotations

from typing import List

CLIENT_UNSAFE_TERMS = [
    "bant",
    "bant coverage",
    "jira",
    "internal",
    "scorecard",
    "coaching",
    "open discovery gap",
    "discovery gaps",
    "discovery coverage",
    "agent envelope",
    "trace id",
    "pod member",
    "is qualifying",
]


def has_client_unsafe_text(text: str) -> bool:
    lowered = (text or "").lower()
    return any(term in lowered for term in CLIENT_UNSAFE_TERMS)


def safe_client_lines(lines: List[str]) -> List[str]:
    out: List[str] = []
    for line in lines or []:
        stripped = (line or "").strip()
        if not stripped:
            continue
        if has_client_unsafe_text(stripped):
            continue
        out.append(stripped)
    return out


def sanitize_client_headline(value: str, account_name: str) -> str:
    text = (value or "").strip()
    if not text or has_client_unsafe_text(text):
        return f"Thank you for your time, {account_name}"
    return text


def sanitize_client_bullets(lines: List[str], fallback: List[str]) -> List[str]:
    safe = safe_client_lines(lines)
    return (safe[:6] if safe else fallback[:6]) or [
        "We discussed your priorities and agreed on next steps for follow-up."
    ]

File: apps/web/src/lib/client-facing-safety.ts
/** Shared client-safe copy rules for emails, CLP, and proposals. */

export const CLIENT_UNSAFE_TERMS = [
  "bant",
  "bant coverage",
  "coverage finished",
  "jira",
  "internal",
  "scorecard",
  "coaching",
  "open discovery gap",
  "open discovery gaps",
  "open gaps",
  "discovery gaps",
  "discovery coverage",
  "decision process",
  "agent envelope",
  "trace id",
  "model:",
  "cost:",
  "a few takeaways",
  "takeaways i captured",
  "call centered on discovery",
  "pod member",
  "is qualifying",
];

export function hasClientUnsafeText(text: string): boolean {
  const lowered = text.toLowerCase();
  return CLIENT_UNSAFE_TERMS.some((term) => lowered.includes(term));
}

export function safeClientLines(lines: string[]): string[] {
  return lines.filter((line) => {
    const lowered = line.toLowerCase();
    return line.trim() && !CLIENT_UNSAFE_TERMS.some((term) => lowered.includes(term));
  });
}

export function sanitizeClientHeadline(value: string, accountName: string): string {
  if (!value.trim() || hasClientUnsafeText(value)) {
    return `Thank you for your time, ${accountName}`;
  }
  return value.trim();
}

export function sanitizeClientBullets(lines: string[], fallback: string[]): string[] {
  const safe = safeClientLines(lines);
  return safe.length ? safe.slice(0, 6) : fallback.slice(0, 6);
}
